contributors/devel/sig-instrumentation/trace-instrumentation.md

```diff
@@ -0,0 +1,75 @@
+## Instrumenting Kubernetes with Traces
+
+The following references and outlines general guidelines for trace instrumentation
+in Kubernetes components. Components are instrumented using the
+[OpenTelemetry Go client library](https://github.com/open-telemetry/opentelemetry-go).
+For non-Go components. [Libraries in other languages](https://opentelemetry.io/docs/languages/)
+are available.
+
+Traces are exposed via gRPC using the [OpenTelemetry Protocol](https://opentelemetry.io/docs/specs/otel/protocol/)
+(OTLP), which is open and well-understood by a wide range of third party
+applications and vendors in the cloud-native eco-system.
+
+The [general instrumentation advice](https://opentelemetry.io/docs/concepts/instrumentation/libraries/)
+from the OpenTelemetry documentation applies. This document reiterates common pitfalls and some
+Kubernetes specific considerations.
+
+### When to instrument
+
+While spans are sampled to avoid high costs, recording too many spans will
+force consumers to lower the sampling rate, and will "drown out" important
+spans. If your component has more than two or three nested spans, you are
+likely over-using trace instrumentation. Most trace instrumentation in
+Kubernetes components falls into one of two categories:
+
+1. Spans for incoming or outgoing network calls
+2. Spans when initiating new work, such as reconciling an object, which may result in network calls.
+
+For network-based telemetry, Kubernetes components should use OpenTelemetry
+instrumentation libraries for
+[HTTP](https://pkg.go.dev/go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp) and
+[gRPC](https://pkg.go.dev/go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc).
```

Comment:
kubelet use library
`go.opentelemetry.io/contrib/instrumentation/github.com/emicklei/go-restful/otelrestful`

https://github.com/kubernetes/kubernetes/blob/227c2e7c2b2c05a9c8b2885460e28e4da25cf558/pkg/kubelet/server/server.go#L337

Should we mention this?

Reply:
I'm OK omitting this.  This is intended to be general guidance, and sometimes many other instrumentation libraries may be needed.